### PR TITLE
Spotbugs: fix error on missing start line value

### PR DESCRIPTION
**Pull request by SPoint42**
Some time, start is not defined and cause an 500 error.
This patch cover the error

### CODE REVIEW

**Comment by valentijnscholten** [comment]
i thinkt the desired solution is to set the line fields to `None` if the linue is not known.

**Comment by SPoint42** [comment]
> i thinkt the desired solution is to set the line fields to `None` if the linue is not known.

But sast_source_line and source_line is an integer in the model....

`sast_source_line = models.IntegerField(` and ` line = models.IntegerField(`

so ?

**Comment by valentijnscholten** [comment]
Just don't set the field and leave it None.

**Comment by valentijnscholten** [comment]
Just don't set the field if there is no value.

**Comment by damiencarol** [comment]
I agree. Better use this structure:

```Python
if 'line' in source_extract and source_extract["line"].isdigit():
  finding.line = int(source_extract["line"])
```

**Comment by damiencarol** [comment]
@SPoint42 any news on this one?
Also could you add a report that break in the PR as unit test ?

File: dojo/tools/spotbugs/parser.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def get_findings(self, filename, test):
             source_extract = bug.find('SourceLine')
             if source_extract is not None:
                 source_file = source_extract.get("sourcepath")
-                source_line = int(source_extract.get("start"))
+                source_line = int(source_extract.get("start") if source_extract.get("start") is not None else "0")
```
Comment (damiencarol) [inline dojo/tools/spotbugs/parser.py:63]:
```suggestion
                if 'start' in source_extract and source_extract["start"].isdigit():
                    source_line = int(source_extract["start"])
```

**Comment by SPoint42** [comment]
> @SPoint42 any news on this one?
Sorry, I was busy on other subject...

> Also could you add a report that break in the PR as unit test ?
will see what I can do with an anonymize report and a unit test.

File: dojo/tools/spotbugs/parser.py (hunk 1 above)
Comment (SPoint42) [reply dojo/tools/spotbugs/parser.py]: Thanks for the quick commit :)

**Comment by damiencarol** [comment]
@SPoint42 just a gentle reminder that we will ship the version `2.0.0`next week. So if by any chance you can finish this PR, it will be shipped.

**Comment by Maffooch** [comment]
@SPoint42 Is this still an active effort?

**Comment by damiencarol** [comment]
Superseded by #5068

**Pull request closed without merging**